crates/sui-network/src/randomness/mod.rs

```diff
@@ -574,18 +580,82 @@ impl RandomnessEventLoop {
         }
 
         debug!("successfully generated randomness full signature");
-        self.completed_sigs.insert((epoch, round));
-        self.remove_partial_sigs_in_range(sig_bounds);
+        self.process_valid_full_signature(epoch, round, sig);
+    }
+
+    #[instrument(level = "debug", skip_all, fields(?peer_id, ?epoch, ?round))]
+    fn receive_full_signature(
```

Comment:
Do we have a test that covers this new flow (e.g., using received full sig), or only implicitly by the e2e tests?

Reply:
added, thanks for the suggestion. also this found a bug